[Gernene Tan] iP

Gernene opened the pull request:
# Duke

Create
- Deadlines
- Events
- Todo Items

- [x] Check off items
- [ ] Mark tasks as not done

1. Fast
2. Simple
3. **Easy**

> I am Duke!

```
public void start(Stage stage) {
        try {
            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
            AnchorPane ap = fxmlLoader.load();
            Scene scene = new Scene(ap);
            scene.getStylesheets().add(Main.class.getResource("/css/style.css").toExternalForm());
            String home = System.getProperty("user.home");
            Path directory = Paths.get(home, "Documents", "duke");
            Path filePath = Paths.get(home, "Documents", "duke", "data.txt");
            duke = new Duke(directory, filePath);
            stage.setScene(scene);
            fxmlLoader.<MainWindow>getController().setDuke(duke);
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
```

`list`

😄 

[Download here](https://github.com/Gernene/ip/releases/tag/v0.2)

Gernene committed:
Complete Level 1

Gernene committed:
Complete level 2

Gernene committed:
Finish level 3

Gernene committed:
Complete level 4

Gernene committed:
Complete level 5

Gernene committed:
Complete level 6

Gernene committed:
Finish level 7

Gernene committed:
Finish level 8

Gernene committed:
Merge branch-Level-8

Gernene committed:
Add more OOP

Gernene committed:
Reorganize code

Gernene committed:
Set up gradle and JUnit

Gernene committed:
Resolve errors

Gernene committed:
Fix gradle problems

Gernene committed:
Enable junit tests

sibinhho99-nus commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,53 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
```
You can add javadoc for public methods

Gernene committed:
Add comments

Gernene committed:
Tweak formatting

Gernene committed:
Find tasks that match keyword

Gernene committed:
Merge branch 'branch-A-CodingStandard'
IP requirement

Gernene committed:
Resolve merge conflicts

Gernene committed:
Add asserts

Gernene committed:
Refactor Parser.java

kerburenthusiasm commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,60 @@
+    return isMarked;
+  }
+
+  public boolean contains(String str) {
```
Perhaps a change to the function name to sound more like a boolean?

```suggestion
  public boolean hasString(String str) {
```

Gernene committed:
Merge pull request #1 from Gernene/branch-A-Assertions

Add asserts

kerburenthusiasm commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,19 @@
+
+public class Todo extends Task {
+
+  public Todo(String description, boolean status) {
```
Perhaps a change to the argument name to sound more like a boolean? I noticed this issue occurring in classes extending task.

```suggestion
  public Todo(String description, boolean isMarked) {
```

Gernene committed:
Merge pull request #2 from Gernene/branch-A-CodeQuality

Refactor Parser.java

kerburenthusiasm commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,23 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task {
+
+  private LocalDate date;
+
+  public Event(String description, boolean status, LocalDate date) {
+    super(description, status);
+    this.date = date;
+  }
+
+  public Event(String description, LocalDate date) {
+    this(description, false, date);
+  }
+
+  public String toString() {
+    return String.format("[E][%s] %s (at: %s)", super.isMarked(), super.getDescription(), DateTimeFormatter.ofPattern("MMM d yyyy"));
+  }
+}
```
Perhaps add header comments for the class and public functions. I noticed the same issue for classes extending Task and Command?

kerburenthusiasm reviewed (commented):
Overall, I found your code easy to read and they follow the Java coding standard provided. However, some files such as Parser.java were difficult to read due to the nested if-else statements. Consider encapsulating some similar statements into a function to improve readability.

Gernene committed:
Fix code standard violations

Gernene committed:
Configure checkstyle

Gernene committed:
Add UI support

Gernene committed:
Add GUI

EvaderFati commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,23 @@
+        this.task = task;
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
Although it is not required, it is recommended to add the '@ Override' annotation. [ https://stackoverflow.com/questions/1005898/should-i-add-an-override-annotation-when-implementing-abstract-methods-in-java](url)

EvaderFati commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,9 @@
+
+public class DukeException extends Exception {
+
+  public DukeException(String message) {
```
According to the **stupid** [Coding standard](https://se-education.org/guides/conventions/java/basic.html#layout), should it be better to indent this line as 4 spaces instead of 2?

EvaderFati reviewed (commented):
Overall your code closely follows the coding standard and is nicely constructed. Only some small writing problems to fix.

Gernene committed:
Fix coding standard violations

Gernene committed:
Fix test coding standard violations

markbiju commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,23 @@
+
+  private LocalDate date;
+
+  public Event(String description, boolean status, LocalDate date) {
```
Perhaps you could name status to be more boolean-sounding? :)

markbiju commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,163 @@
+     * @return UnmarkCommand instance.
+     */
+    public Command parseUnmarkCommand(String fullCommand) {
+        String dataString = fullCommand.replaceFirst("unmark ", "");
```
Commendable discretion shown here to differentiate when to use camelCase and when to use PascalCase.

markbiju commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,64 @@
+                assert data[1].equals("true") || data[1].equals("false") : throw DukeException("Invalid file input");
+                Boolean status = Boolean.parseBoolean(data[1]);
+                String text = data[2];
+                if (data[0].equals("T")) {
```
The if-else class of statements should have the following form:
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}

markbiju commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+  }
+
+  public String toString() {
+    return String.format("[D][%s] %s (by: %s)", super.isMarked(), super.getDescription(), DateTimeFormatter.ofPattern("MMM d yyyy"));
```
Perhaps this line is rather long and would benefit from a line break?

markbiju reviewed (commented):
Excellent attempt at following coding standard with absolutely minimal coding standard violations

Gernene committed:
Display UI

Gernene committed:
Merge pull request #3 from Gernene/branch-wk5

Branch wk5

Gernene committed:
Add contacts

Gernene committed:
Fix bugs

Gernene committed:
Save list to harddisk

Gernene committed:
Style GUI

Gernene committed:
Refine unit tests for Parser and Storage

Unit tests for Parser and Storage are incomplete.

Adding more test methods helps ensure code quality.

Let's add unit tests methods to ensure storage and command line data is
interpreted correctly.

Gernene committed:
Create user guide

Gernene committed:
Merge pull request #5 from Gernene/polish

Prepare duke for release

Gernene committed:
Polish minor details

Gernene committed:
Fix text wrap on dialog box

Gernene committed:
Merge pull request #6 from Gernene/polish

Polish

Gernene committed:
Fix text-wrapping issue

Gernene committed:
Exit UI on command

Gernene committed:
Merge pull request #7 from Gernene/polish

Fix dialog text wrapping

Gernene committed:
Update build.gradle for win compatibility

Gernene committed:
Improve contacts functionality

Gernene committed:
Improve error handling

Gernene committed:
Update GUI

Gernene committed:
Change button color

Gernene committed:
Update UG

Gernene committed:
Merge pull request #9 from Gernene/polish

Polish
